server/Arcadia.Assistant.Web/Controllers/PushNotificationsController.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace Arcadia.Assistant.Web.Controllers
+{
+    using System.Threading;
+    using System.Threading.Tasks;
+
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    using Arcadia.Assistant.Notifications.Push;
+    using Arcadia.Assistant.Web.Authorization;
+    using Arcadia.Assistant.Web.PushNotifications;
+    using Arcadia.Assistant.Web.Users;
+
+    [Route("/api/push/device")]
+    [Authorize(Policies.UserIsEmployee)]
+    public class PushNotificationsController : Controller
+    {
+        private readonly IPushNotificationsService pushNotificationsService;
+        private readonly IUserEmployeeSearch userEmployeeSearch;
+
+        public PushNotificationsController(
+            IPushNotificationsService pushNotificationsService,
+            IUserEmployeeSearch userEmployeeSearch)
+        {
+            this.pushNotificationsService = pushNotificationsService;
+            this.userEmployeeSearch = userEmployeeSearch;
+        }
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status202Accepted)]
+        public async Task<IActionResult> RegisterDevice([FromBody]string deviceId, CancellationToken cancellationToken)
+        {
+            var employee = await this.userEmployeeSearch.FindOrDefaultAsync(this.User, cancellationToken);
+
+            await this.pushNotificationsService.RegisterDevice(employee.Metadata.EmployeeId, deviceId, cancellationToken);
+
+            return this.Accepted();
+        }
+
+        [HttpDelete]
```

Comment:
I believe the following interface is better:
PUT /api/push/device/{deviceId}
DELETE /api/push/device/{deviceId}
At least because AjaxObservable wrapper doesn't accept body for delete. :)

Reply:
Ok, will fix this.

Reply:
But I suppose POST fits specification more, because POST to `/device/{deviceId}` creates new device id for current user, not updates existing one. But from the other side, it updates current user devices :)

Reply:
AFAIK, PUT and DELETE usually come together. But it's up to you, we can use both.

Reply:
@sponkrashin I would avoid using 'deviceId', in mobile world it has its own semantic. It would be better to have it like 'appCenterInstallId', cause this is exactly what are we going to pass.

Reply:
Well, let it be 'devicePushToken' then, cause on our case appCenterInstallId incapsulates both APNs Device Token and GCM Registration ID/Firebase Token.